pandas/tests/groupby/test_function.py

```diff
@@ -459,35 +460,33 @@ def test_groupby_cumprod():
     tm.assert_series_equal(actual, expected)
 
 
-def test_ops_general():
-    ops = [('mean', np.mean),
-           ('median', np.median),
-           ('std', np.std),
-           ('var', np.var),
-           ('sum', np.sum),
-           ('prod', np.prod),
-           ('min', np.min),
-           ('max', np.max),
-           ('first', lambda x: x.iloc[0]),
-           ('last', lambda x: x.iloc[-1]),
-           ('count', np.size), ]
-    try:
-        from scipy.stats import sem
-    except ImportError:
-        pass
-    else:
-        ops.append(('sem', sem))
+def scipy_sem(*args, **kwargs):
+    from scipy.stats import sem
+    return sem(*args, ddof=1, **kwargs)
+
+
+@pytest.mark.parametrize(
+    'op,targop',
+    [('mean', np.mean),
+     ('median', np.median),
+     ('std', np.std),
+     ('var', np.var),
+     ('sum', np.sum),
+     ('prod', np.prod),
+     ('min', np.min),
+     ('max', np.max),
+     ('first', lambda x: x.iloc[0]),
+     ('last', lambda x: x.iloc[-1]),
+     ('count', np.size),
+     pytest.param(
+         'sem', scipy_sem, marks=td.skip_if_no_scipy)])
+def test_ops_general(op, targop):
```

Comment:
Might be overlooking but not immediately obvious how this tests the preservation of Categorical and Spares types - is this just a general cleanup?

Reply:
this is just general cleaning

Reply:
So do we actually have test coverage for the preservation of Categorical / Sparse in groupby tests?

Reply:
https://github.com/pandas-dev/pandas/pull/27071/files#diff-e88b6d0523a37f080cc5b64b9e6f24f7L285

Reply:
Can we either repurpose this test or maybe add a new test for all of the ops and the preservation of sparse / categorical? Looks like the one you linked to only does first / last for Categorical, so a small subset of what is promised here

Reply:
actually added the OP tests now.

Reply:
Reading my mind...